File: poke_center_time_conversions-10/solution.cpp
#include <iostream>
#include <iomanip>

using namespace std;

int main() {
  int count, sec, min, hour, tmp;
  cin >> count;
  for(int i = 0; i < count; i++){
    cin >> tmp;
    sec = tmp % 100;
    tmp /= 100;
    min = tmp % 100;
    tmp /= 100;
    hour = tmp;
    if(hour > 12 && hour != 24){
      cout << setfill('0') << setw(2) <<  (hour-12) << ":" << setw(2) << min << ":" << setw(2) << sec<< " PM" << endl;
    } else if ( hour == 00 || hour == 24){
      cout << setfill('0') << setw(2) <<  12 << ":" << setw(2) << min << ":" << setw(2) << sec << " AM" << endl;
    } else {
      cout << setfill('0') << setw(2) <<  hour << ":" << setw(2) << min << ":" << setw(2) << sec << " AM" << endl;
    }
  }
  return 0;
}

File: poke_center_time_conversions-10/solution.py
#!/usr/bin/python

def buff(x):
    tmp = "0" + str(x)
    return tmp[-2:]

count = int(raw_input())

for x in xrange(0, count):
    tmp = raw_input()
    hr = int(tmp[0:2])
    min= tmp[2:4]
    sec= tmp[4:6]
    if hr > 12 and hr != 24:
        print(buff(hr-12) + ":" + min + ":" + sec + " PM")
    elif hr == 00 or hr == 24:
        print("12" + ":" + min + ":" + sec + " AM")
    else:
        print(buff(hr) + ":" + min + ":" + sec + " AM")